.github/scripts/platform_lib.py

```diff
@@ -244,10 +249,17 @@ def change_workflow_instance_states(self, gh_token: str, workflow_tag: str, stat
         elif state_change == 'terminate':
             print(f"Terminating instances: {', '.join(instance_ids)}")
             self.deregister_runners(gh_token, f"aws-{workflow_tag}")
-            self.client.terminate_instances(InstanceIds=instance_ids, DryRun=dryrun)
+            self.platform_terminate_instances(instance_ids, dryrun)
         else:
             raise ValueError(f"Unrecognized transition type: {state_change}")
 
+    def platform_terminate_instances(self, platform_list: List[Any], dryrun: bool = False) -> None:
```

Comment:
So the reason this is done is to avoid some of the typing issues with the boto3 client?

Reply:
In the `cull_instances` file you explicitly call the `terminate_azure_vms` which is a hidden method in the platform library. So I added this new method so that we avoid piping platform-specific methods out from the platform lib.